src/essreflectometry/amor/orso.py

```diff
@@ -1,60 +1,95 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-import platform
-from datetime import datetime
+"""ORSO utilities for Amor."""
+import numpy as np
+import scipp as sc
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source as orso_data_source
+from orsopy.fileio.orso import Column, Orso, OrsoDataset
 
-from orsopy import fileio
+from ..orso import OrsoDataSource, OrsoInstrument, OrsoIofQDataset, OrsoReduction
+from ..types import NormalizedIofQ1D, QResolution, Run, ThetaData, WavelengthData
 
-from .. import __version__
 
+def build_orso_instrument(
+    events_in_wavelength: WavelengthData[Run], events_in_theta: ThetaData[Run]
+) -> OrsoInstrument[Run]:
+    """Build ORSO instrument metadata from intermediate reduction results for Amor.
 
-def make_orso(
-    owner: fileio.base.Person,
-    sample: fileio.data_source.Sample,
-    creator: fileio.base.Person,
-    reduction_script: str,
-) -> fileio.orso.Orso:
+    This assumes specular reflection and sets the incident angle equal to the computed
+    scattering angle.
     """
-    Generate the base Orso object for the Amor instrument.
-    Populate the Orso object for metadata storage.
-
-    Parameters
-    ----------
-    owner:
-        The owner of the data set, i.e. the main proposer of the measurement.
-    sample:
-        A description of the sample.
-    creator:
-        The creator of the reduced data, the person responsible for the
-        reduction process.
-    reduction_script:
-        The script or notebook used for reduction.
-
-    Returns
-    -------
-    :
-        Orso object with the default parameters for the Amor instrument.
-    """
-    orso = fileio.orso.Orso.empty()
-    orso.data_source.experiment.probe = 'neutrons'
-    orso.data_source.experiment.facility = 'Paul Scherrer Institut'
-    orso.data_source.measurement.scheme = 'angle- and energy-dispersive'
-    orso.reduction.software = fileio.reduction.Software(
-        'scipp-ess', __version__, platform.platform()
+    wavelength = events_in_wavelength.coords['wavelength']
```

Comment:
This assumes that there are wavelength coords, which may not be the case if we do not start with a TOF coords, or do not have wavelength masks. If not found, fall back to `events_in_wavelength.bins.coords['wavelength']`.